### PR TITLE
Add logger template

### PR DESCRIPTION
**Pull request by Scorpiour**

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/logger/HugeGraphLogger.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    /**
+     * Allow alternate level
+     * @param level customized level
+     * @param template LogTemplate
+     * @param args Corresponding args
+     * @param <T> Level Type
+     */
+    public <T extends GraphLogLevel> void log(Class<T> level, LogTemplate template, Object ...args) {
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/logger/HugeGraphLogger.java]: not exceed 80 chars per line

File: src/main/java/com/baidu/hugegraph/logger/HugeGraphLogger.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public void logApiAccess(String method, String url, Long requestTime) {
+        infoLogger.generalLogMessage(LogTemplate.ACCESS_LOG, method, url, requestTime);
+    }
+
+    public void logRestServerStart() {
+        infoLogger.generalLogMessage(LogTemplate.START_REST_SERVER);
+    }
+
+    public void logServerShutdown() {
+
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/logger/HugeGraphLogger.java:195]: no output when shutdown?

File: src/main/java/com/baidu/hugegraph/logger/HugeGraphLogger.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/logger/HugeGraphLogger.java:199]: delete empty line

File: src/main/java/com/baidu/hugegraph/logger/LogTemplate.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Map;
+
+/**
+ * LogTemplate is used to generate preset logs
+ * @since 2021-11-25
+ */
+public enum LogTemplate {
+    UNKNOWN("UNKNOWN", ""),
+    ACCESS_LOG("ACCESS_LOG",
+            "{} /{} Status: {} - user: {} {} - roles: {} in {} ms"),
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/logger/LogTemplate.java]: align with "ACCESS_LOG"

File: src/main/java/com/baidu/hugegraph/logger/LogTemplate.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    START_REST_SERVER("START_REST_SERVER", "restServer has been initialized!"),
+    ;
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/logger/LogTemplate.java:360]: add empty line

File: src/main/java/com/baidu/hugegraph/logger/LogTemplate.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * LogTemplate is used to generate preset logs
+ * @since 2021-11-25
+ */
+public enum LogTemplate {
+    UNKNOWN("UNKNOWN", ""),
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/logger/LogTemplate.java:32]: Just 3 templates? How to add template in the future if needed?

**Review by zhoney — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/concurrent/AtomicLock.java [hunk 7]
```diff
@@ -21,14 +21,14 @@
...
-import org.slf4j.Logger;
-
+import com.baidu.hugegraph.logger.HugeGraphLogger;
 import com.baidu.hugegraph.util.Log;
 
 public class AtomicLock {
 
-    private static final Logger LOG = Log.logger(LockManager.class);
-
+    // private static final Logger LOG = Log.logger(LockManager.class);
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/concurrent/AtomicLock.java:29]: delete

File: src/main/java/com/baidu/hugegraph/logger/AuditLogger.java [hunk 8]
```diff
@@ -0,0 +1,548 @@
...
+     * @param instanceId
+     * @param executorId
+     */
+    public void logExportInstanceLog(String instanceId, String executorId) {
+        infoLogger
+        .generalLogMessage(LogTemplate.EXPORT_INSTANCE_LOG, executorId);
+    }
+
+    /**** Host related logs ****/
+
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/logger/AuditLogger.java:559]: will add corresponding log in the future？

File: src/main/java/com/baidu/hugegraph/concurrent/AtomicLock.java [hunk 9]
```diff
@@ -21,14 +21,14 @@
 
 import java.util.concurrent.atomic.AtomicReference;
 
-import org.slf4j.Logger;
-
+import com.baidu.hugegraph.logger.HugeGraphLogger;
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/concurrent/AtomicLock.java:24]: prefer also add logger for hugegraph-client

**Review by zhoney — CHANGES_REQUESTED** [review]

File: src/main/java/com/baidu/hugegraph/logger/AuditLogger.java (hunk 8 above)
Comment (Scorpiour) [reply src/main/java/com/baidu/hugegraph/logger/AuditLogger.java:559]: yes

**Review by zhoney — APPROVED** [review]

**Pull request merged**